### ISSUE TITLE
Only basic thermostats seem to be supported

### ISSUE THREAD

**Issue by DrMWeigand** [pgerke/homebridge-freeathome-local-api#74]
**Describe The Bug:**
Not all free@home thermostats are added to Home app

**To Reproduce:**
Try to add free@home system with basic and comfort thermostats

**Expected behavior:**
All thermostats should be displayed and  be usable

**Plugin Config:**

```json
            "name": "xxx",
            "host": "192.168.x.x",
            "user": "xxx",
            "password": "xxx",
            "tlsEnabled": false,
            "disableCertificateVerification": false,
            "motionSensorAutoReset": true,
            "motionSensorDefaultResetTimer": 20000,
            "verboseErrors": false,
            "experimental": true,
            "platform": "FreeAtHomeLocalApi"
        }
```

**Screenshots:**

Using mixed config of basic and comfort thermostats
<img width="379" alt="Bildschirmfoto 2023-07-26 um 19 30 17" src="https://github.com/pgerke/homebridge-freeathome-local-api/assets/89643686/59fe98c4-889b-45b1-ba0e-42643e98cdbd">

Only basic thermostat is displayed
<img width="1151" alt="Bildschirmfoto 2023-07-26 um 19 31 34" src="https://github.com/pgerke/homebridge-freeathome-local-api/assets/89643686/8335f8cb-a6ef-4399-ad98-150021fda42a">

-> Unknown door opener devices are displayed instead

**Environment:**

* **Plugin Version**: v1.6.0
* **Homebridge Version**: v1.6.0
* **Node.js Version**: v18.12.1
* **Operating System**: Raspbian

**Comment by pgerke** [comment]
At the moment the plugin supports room temperature controllers with the function ID `23`. I suspect the Basic and Comfort RTCs are new devices that have a different function ID and possibly different pairing IDs, channels, data points... @DrMWeigand, could you kindly attach a device config file so I can confirm?

The door openers come from the welcome panel and are unrelated to the thermostats.

**Comment by DrMWeigand** [comment]
Thank you for the fast response. What do you mean with the device config file? The B+J system was not implemented by myself but by an electrician. I don’t have much experience with the free at home system itself and just have sufficient rights for using the API and administrate devices.

**Comment by pgerke** [comment]
My plugin communicates with the System Access Point using the local API REST interface. Each device type has a unique function ID and devices with the same function ID use the same data points to exchange information. The reason the (probably new) 'Comfort' type of room temperature controllers don't show up in Apple Home is, that it (most likely) has a different function ID compared to the "normal" ones that the plugin does not know. If a device is reported by the SysAP, that the plugin does not recognize, it ignores it.

To support a new device type I need to know the data points on which the device provides data and expects updates. And for that I need a sample configuration from the device. I have created a guide (here: https://github.com/pgerke/homebridge-freeathome-local-api/wiki/Retrieving-the-Configuration-for-a-Device-from-the-System-Access-Point) as to how retrieve a device configuration from the System Access Point. If you provide the configuration, I can try and support the new type in the plugin.

**Comment by DrMWeigand** [comment]
Sorry for the late answer. I was on vacation and somehow lost track of this. I just retrieved the data as given by your instructions. The function id of the comfort thermostats seems to be 3e. You'll find the .json response for three thermostats that I could unambiguously identify as comfort thermostats attached. Looking forward to the integration of this device type 😊
[response_1691572809037.txt](https://github.com/pgerke/homebridge-freeathome-local-api/files/12300586/response_1691572809037.txt)
[response_1691572790383.txt](https://github.com/pgerke/homebridge-freeathome-local-api/files/12300587/response_1691572790383.txt)
[response_1691572671240.txt](https://github.com/pgerke/homebridge-freeathome-local-api/files/12300588/response_1691572671240.txt)

**Comment by pgerke** [comment]
Thanks, that seems to be everything I need. I'll let you know once I have a preview version available for testing.

**Comment by pgerke** [comment]
I've published a preview version [v1.7.0-pre.1](https://www.npmjs.com/package/homebridge-freeathome-local-api/v/1.7.0-pre.1) that adds support for the new device type as an experimental accessory. Please feel free to check it out and provide feedback! If there are no issues I'll release the change officially shortly.

**Comment by pgerke** [comment]
@DrMWeigand any feedback yet?

**Comment by DrMWeigand** [comment]
I've just installed the v1.7.0-pre.1. Thermostats are now displayed in the Home app and set temperatures on/off state are instantly reflected in the Busch Jäger web-interface. Measured temperatures seem also to be plausible but it's harder to compare since in the Busch Jäger system the mean temperature from multiple thermostats seem to be taken. Only caveat is that for each new thermostat an error message was displayed which was looking like this: 

[26/08/2023, 13:19:10] [FreeAtHomeLangstreet] Adding new accessory: Hzg Küche
[26/08/2023, 13:19:10] [FreeAtHomeLangstreet] ABB700D78F22 (ch0000): Cannot configure accessory for FunctionID '3e'!

Thank you very much for your work!

**Comment by pgerke** [comment]
> Measured temperatures seem also to be plausible but it's harder to compare since in the Busch Jäger system the mean temperature from multiple thermostats seem to be taken.

Yes, I have noted that myself. I have a room that has two B+J room temperature controllers. in the free@home app I'm seeing an (incorrect) aggregated temperature. 

> [26/08/2023, 13:19:10] [FreeAtHomeLangstreet] ABB700D78F22 (ch0000): Cannot configure accessory for FunctionID '3e'!

Thanks for that hint. I'll publish a new version shortly that fixes this issue.

**Comment by pgerke** [comment]
@DrMWeigand The config error message should no longer appear in [v1.7.0-pre.2](https://www.npmjs.com/package/homebridge-freeathome-local-api/v/1.7.0-pre.2). If you can confirm and don't experience any further issues I'll prepare the release.

**Comment by DrMWeigand** [comment]
@pgerke I can confirm that the issue with the error is resolved. I tested again to toggle the thermostats and setting different temperatures. Everything seemed to work i.e. was visible in the homebridge log and reflected in the Busch-Jäger web application. Thank you again!

**Comment by pgerke** [comment]
@DrMWeigand Thanks for the quick feedback, I have published the feature officially with [v1.7.0](https://www.npmjs.com/package/homebridge-freeathome-local-api/v/1.7.0).
